File: src/main/java/plugin/interaction/combat/magic/ancient/teleport/AnnakarlTeleportSpell.java
package plugin.interaction.combat.magic.ancient.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class AnnakarlTeleportSpell implements TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 60;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(LAW_RUNE, 2, BLOOD_RUNE, 2);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(3288, 3886, 0);
	}
	
	@Override
	public int spellId() {
		return 46;
	}
	
	@Override
	public double exp() {
		return 100;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.ANCIENTS;
	}
}

File: src/main/kotlin/plugin/interaction/combat/magic/ancient/teleport/AnnakarlTeleportSpell.kt
package plugin.interaction.combat.magic.ancient.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/27/2017
 */
class AnnakarlTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 60
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.LAW_RUNE, 2, MagicConstants.BLOOD_RUNE, 2)
    }

    override fun destination(): WorldTile {
        return WorldTile(3288, 3886, 0)
    }

    override fun spellId(): Int {
        return 46
    }

    override fun exp(): Double {
        return 100.0
    }

    override fun book(): MagicBook {
        return MagicBook.ANCIENTS
    }
}